users/vieting/experiments/librispeech/librispeech_100_ctc/fairseq_finetuning/config_01_finetune.py

```diff
@@ -0,0 +1,185 @@
+            #"max_sample_size": 20 * sample_rate,  # max 20s, length of one crop in batch
+            #"min_sample_size": 2 * sample_rate,  # 2s of minimal length
+            #"sample_rate": sample_rate,
+            "normalize": False,
+            "labels": "ltr"
+        },
+        "dataset": {
+            "num_workers": 6,
+            "max_tokens": 3200000,  # length of tokens in one batch
+            "skip_invalid_size_inputs_valid_test": True,
+            "valid_subset": "valid",
+        },
+        "distributed_training": {
+            "distributed_world_size": num_gpus,
+            "ddp_backend": "legacy_ddp"
+        },
+        "criterion": {
+            "_name": "ctc",
+            "zero_infinity": True,
+            # TODO: check if this is needed
+            #"infonce": True,
+            #"log_keys": ["prob_perplexity", "code_perplexity", "temp"],
+            #"loss_weights": [0.1, 10]
+        },
+        "optimization": {
+            "sentence_avg": True,
+            "max_update": 80000,
+            "lr": [0.00003],
+            "update_freq": [8 // num_gpus],
+            # TODO check if this is needed
+            #"max_epoch": 1000,
+        },
+        "optimizer": {
+            "_name": "adam",
+            "adam_betas": "(0.9,0.98)",
+            "adam_eps": "1e-08",
+            # TODO check if this is needed
+            #"weight_decay": 0.01,
+        },
+        "lr_scheduler": {
+            "_name": "tri_stage",
+            "phase_ratio": [0.1, 0.4, 0.5],
+            "final_lr_scale": 0.05,
+        },
+        "model": {
+            "_name": "wav2vec_ctc",
+            "w2v_path": w2v_path,
+            "apply_mask": True,
+            "mask_prob": 0.65,
+            "mask_channel_prob": 0.5,
+            "mask_channel_length": 64,
+            "layerdrop": 0.1,
+            "activation_dropout": 0.1,
+            "feature_grad_mult": 0.0,
+            "freeze_finetune_updates": 0,
+        }
+    }
+    return fairseq_args
+
+
+def main():
+    prefix_name = "experiments/librispeech/librispeech_100_ctc/fairseq/"
+    # run pre-training
+    exp_name = "base"
+    num_gpus = 2
+    gpu_mem_rqmt = 24
+    corpus_names = ["train-clean-100"]
+    #python_path = "work/asr3/vieting/hiwis/pletschko/miniconda3/envs/fairseq_python38/bin/python"
+    #setattr(gs, "FAIRSEQ_PYTHON_EXE", python_path)
+
+    fairseq_args = get_fairseq_args(corpus_names=corpus_names, num_gpus=num_gpus, w2v_path="/work/asr3/vieting/hiwis/pletschko/fairseq/models/wav2vec_small.pt")
+    fairseq_config = FairseqHydraConfig(fairseq_args)
+    fairseq_root = get_fairseq_root()
+    job = FairseqHydraTrainingJob(
+        fairseq_config,
+        max_epoch=300,
+        save_interval=25,
+        time_rqmt=1,
+        mem_rqmt=8,
+        cpu_rqmt=2,
+        gpu_rqmt=num_gpus,
+        gpu_mem_rqmt=gpu_mem_rqmt,
+        fairseq_root=fairseq_root,
+        fairseq_python_exe=python_path,
+        use_cache_manager=False,
```

Comment:
We should change this and make sure that we cache files properly to avoid high loads on the file server. We might discuss this offline if you're unsure how to do/check that.

Reply:
I changed this to `True`, however the caching of the individual files takes quite long (about 1h 45min for LS100). While this works, we might want to look into using zipped directories.